riscv/sim.cc

```diff
@@ -128,7 +118,7 @@ sim_t::sim_t(const cfg_t *cfg, const char* varch, bool halted, bool real_time_cl
         procs[cpu_idx]->set_pmp_num(pmp_num);
       } else {
         std::cerr << "core ("
-                  << hartids.size()
+                  << cfg->hartids().size()
```

Comment:
How does this work? (How did it ever work?) If I have hartids={0,1} this would report that core 2 has invalid pmpregions?